### PR TITLE
Add extend filter to REST API

### PR DESCRIPTION
**Pull request by szok**
Added extend_filter_fields to rest framework api

### CODE REVIEW

File: src/ralph/api/viewsets.py [hunk 1]
```diff
@@ -116,4 +116,15 @@ class RalphAPIViewSet(
...
-    pass
+    extend_filter_fields = {}
+
+    def get_queryset(self):
+        queryset = self.queryset
+        for field, field_filters in self.extend_filter_fields.items():
+            value = self.request.query_params.get(field, None)
+            if value:
+                queryset = queryset.filter(
+                    **dict.fromkeys(field_filters, value)
```
Comment (mkurek) [inline src/ralph/api/viewsets.py]: shouldn't this use `Q` to make or-condition at the level of single param?

File: src/ralph/api/viewsets.py [hunk 2]
```diff
@@ -116,4 +119,15 @@ class RalphAPIViewSet(
...
-    pass
+    extend_filter_fields = {}
+
+    def get_queryset(self):
+        queryset = self.queryset
+        qs = []
+        for field, field_filters in self.extend_filter_fields.items():
+            value = self.request.query_params.get(field, None)
+            if value:
+                qs.append(Q(**dict.fromkeys(field_filters, value)))
```
Comment (mkurek) [inline src/ralph/api/viewsets.py]: use AND between fields (meta-fields) and OR within one field

File: src/ralph/api/viewsets.py [hunk 3]
```diff
@@ -116,4 +117,15 @@ class RalphAPIViewSet(
     viewsets.ModelViewSet,
     metaclass=RalphAPIViewSetMetaclass
 ):
-    pass
+    extend_filter_fields = {}
```
Comment (mkurek) [inline src/ralph/api/viewsets.py]: this should probably be done in `__init__`

File: src/ralph/api/viewsets.py [hunk 4]
```diff
@@ -116,4 +117,20 @@ class RalphAPIViewSet(
...
-    pass
+    extend_filter_fields = None
+
+    def __init__(self, *args, **kwargs):
+        if self.extend_filter_fields is None:
+            self.extend_filter_fields = {}
+        super().__init__(*args, **kwargs)
+
+    def get_queryset(self):
+        queryset = self.queryset
```
Comment (mkurek) [inline src/ralph/api/viewsets.py]: why not super?

**Comment by mkurek** [comment]
uni tests would be nice here :)

**Comment by mkurek** [comment]
:ok:

**Comment by ar4s** [comment]
:ok:

**Pull request merged**